nemo/collections/asr/modules/graph_decoder.py

```diff
@@ -0,0 +1,108 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+
+from nemo.core.classes import NeuralModule
+from nemo.core.neural_types import LengthsType, LogprobsType, NeuralType, PredictionsType
+
+
+class ViterbiDecoderWithGraph(NeuralModule):
+    @property
+    def input_types(self):
+        """Returns definitions of module input ports.
+        """
+        return {
+            "log_probs": NeuralType(("B", "T", "D"), LogprobsType()),
+            "log_probs_length": NeuralType(tuple("B"), LengthsType()),
+        }
+
+    @property
+    def output_types(self):
+        """Returns definitions of module output ports.
+        """
+        return {"predictions": NeuralType(("B", "T"), PredictionsType())}
+
+    def __init__(
+        self,
+        num_classes,
+        backend="k2",
```

Comment:
Put types for all arguments to methods, wherever feasible. `backend: str = "k2"` etc